experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -546,9 +563,11 @@ const mapStaticDispatchMap: PropertyDescriptorMap = {
 		},
 	},
 	entries: {
-		value(this: SharedTreeMap<MapSchema>): IterableIterator<[string, unknown]> {
+		*value(this: SharedTreeMap<MapSchema>): IterableIterator<[string, unknown]> {
```

Comment:
@Josmithr - `entries', 'values' and iteration forgot to proxify their output. They were returning TreeNodes directly. I've fixed it here since my test uses `values` to iterate its maps. If you want to fix this in a separate PR to get the fix in sooner, go ahead, otherwise I will check it in as part of this PR.